### ISSUE TITLE
(Fractional) scaling issues on KDE/Wayland

### ISSUE THREAD

**Issue by major-mayer** [dbeaver/dbeaver#26846]
### Description

DBeaver looks like this when launching it on my instance:
![grafik](https://github.com/dbeaver/dbeaver/assets/17907799/541ba668-2dd1-4e59-8431-bcc222b3aec4)
I am using KDE 5.27.11 and Wayland with fractional scaling set to 140% on a UHD display.
But even if I set the display scaling to 200%, so a non-fractional scaling value, the UI is very distorted, just that now everything is a bit larger. 

Since [this](https://github.com/dbeaver/dbeaver/issues/6581) issue was closed because apparently DBeaver looks correct now on Gnome/ Wayland, I thought it might be time to open a new one specifically for KDE/ Wayland.

Workaround: Start DBeaver using XWayland `GDK_BACKEND=x11 dbeaver`.
The problem with this workaround is that no fractional scaling is supported for x11, so `GDK_SCALE=1.4` wouldn't work (I think it's rounded to 1.0).
So all UI elements are very small on a HiDPI display.

### DBeaver Version

Community Version 24.0.0.202403042157

### Operating System

Manjaro Linux KDE/ Wayland

### Database and driver

Not relevant

### Steps to reproduce

1. Set up a KDE Wayland session
2. Start DBeaver 
3. Notice the numerous UI glitches

### Additional context

Operating System: Manjaro Linux 
KDE Plasma Version: 5.27.11
KDE Frameworks Version: 5.115.0
Qt Version: 5.15.12
Kernel Version: 6.6.26-1-MANJARO (64-bit)
Graphics Platform: Wayland
Processors: 6 × Intel® Core™ i5-8600K CPU @ 3.60GHz
Memory: 31.3 GiB of RAM
Graphics Processor: AMD Radeon RX 6800
Manufacturer: Micro-Star International Co., Ltd.
Product Name: MS-7B46
System Version: 1.0

**Comment by anaximeno** [comment]
I also noticed the same issue while testing **Fedora KDE 40** which was released with the new **KDE Plasma 6**. Also if you have multi-monitors with different scales it will ask you to restart the app every time you move the dbeaver window to another display.

**Comment by Lanpingner** [comment]
+1
I am facing similar issues since i updated my wl-roots for the the UI sometime is dissapier as well or if i start it with 
`GDK_BACKEND=x11` the clipboard copies HTML

**Comment by flambchap** [comment]
Hi guys, DBeaver works perfectly fine for me on a 4K display. It seems like some environment detail between our systems causes the issue. Do you have any ideas about what it might be?

![Screenshot_20240725_135416](https://github.com/user-attachments/assets/05219611-728e-4024-8ebc-4264db1bbfdd)

**Comment by flambchap** [comment]
By the way, does clicking Help > Reset Settings... > Apply and Restart fix the issue?

**Comment by gc-pwr** [comment]
> By the way, does clicking Help > Reset Settings... > Apply and Restart fix the issue?

i got the same behavior on arch linux 
kde plasma: 6.1.3
kde framework: 6.4.0
kernel version: 6.10.0-arch1-2
Wayland

Reset settings - apply and restart doesn't fix the issue

**Comment by anaximeno** [comment]
> Hi guys, DBeaver works perfectly fine for me on a 4K display. It seems like some environment detail between our systems causes the issue. Do you have any ideas about what it might be?

Is it an hdpi display? The issue seems to be with scaling between 100% and 200%.

**Comment by flambchap** [comment]
>Is it an hdpi display? The issue seems to be with scaling between 100% and 200%.

Yes, it's a HiDPI display, and I'm using fractional scaling (160%).

**Comment by Cris70** [comment]
I have the same problem here.
openSuse Tumbleweed, fractional scaling at 150%

Operating System: openSUSE Tumbleweed 20241211
KDE Plasma Version: 6.2.4
KDE Frameworks Version: 6.8.0
Qt Version: 6.8.1
Kernel Version: 6.11.8-1-default (64-bit)
Graphics Platform: Wayland

Processors: 12 × AMD Ryzen 5 3600 6-Core Processor
Memory: 31,3 GiB of RAM
Graphics Processor: AMD Radeon RX 550 / 550 Series
Manufacturer: Micro-Star International Co., Ltd.
Product Name: MS-7C56
System Version: 1.0

**Comment by LeoPazEs** [comment]
I got black boxes on Wayland with an ultrawide display. 
Reset settings - apply and restart doesn't fix the issue

**Comment by Cris70** [comment]
The only "fix" for me is to start dbeaver with these environment variables: 
- GDK_BACKEND=x11 
- GDK_DPI_SCALE=1.5

**Comment by LeoPazEs** [comment]
> The only "fix" for me is to start dbeaver with these environment variables:
> 
> * GDK_BACKEND=x11
> * GDK_DPI_SCALE=1.5

In my case I only need GDK_BACKEND

**Comment by metalgrid** [comment]
Same issue using other Wayland compositors (niri or sway - identical behavior). Using `GDK_BACKEND=x11` fixes the UI, but everything is tiny. Combining with `GDK_DPI_SCALE=2`, fonts are displayed correctly, but icons are left seemingly with no scaling, so the UI gets distorted:

<img width="1410" height="107" alt="Image" src="https://github.com/user-attachments/assets/7ba0d054-829d-47e9-92ee-40ba8d04b3d4" />

Another specific behavior is that if I launch DBeaver straight from the shortcut (no environment variables at all) on a second display with no scaling and then move it to the HiDPI display, everything works correctly (there's a dialog "Restart for the zoom changes to take full effect?" to which I select "Don't restart"). If I can provide any other information that could help, please let me know!

**Comment by anthops** [comment]
I have the same issue as you @metalgrid. Icons are super small after setting `GDK_DPI_SCALE=1.5`

<img width="1781" height="445" alt="Image" src="https://github.com/user-attachments/assets/3afeaf48-e047-4f2a-9fbc-a02e2079b359" />

**Comment by FKPSC** [comment]
It seems to happen when multiple monitors are involved. Related: #37941

**Comment by sebastianfrey** [comment]
I have the same issue with Ubuntu 25.10. My primary monitor is a 4k monitor with fractional scaling set to 125%. After starting DBeaver the window looks as follows:

<img width="1355" height="676" alt="Image" src="https://github.com/user-attachments/assets/4ff26256-93f8-496d-b45b-114aa34f5a32" />

After I drag the DBeaver window either to the integrated laptop screen or my second 2k monitor, both without fractional scaling, the DBeaver UI comes back to live:
 
<img width="1336" height="667" alt="Image" src="https://github.com/user-attachments/assets/5e0f8b2a-edd2-4216-b509-69a8e9f08720" />

When I than drag the DBeaver window back to my primary monitor, it somehow detects a zoom change, and want to restart:

<img width="1336" height="662" alt="Image" src="https://github.com/user-attachments/assets/bd28ba53-962e-42bb-91a7-dec1033b3d9b" />


If you click `Don't restart`, the DBeaver window remains usable. If you click `Restart` it breaks again. For me the popup shows up every time I drag DBeaver from my second monitor to my primary monitor with fractional scaling enabled.

So if you have a second monitor, without fractional scaling, dragging the window might be a workaround.

**Comment by maciejopalinski** [comment]
I have severe Wayland fatigue... IT'S ALWAYS WAYLAND

**Comment by sabercoy** [comment]
I came to report this bug for Fedora 44 GNOME 50 (Wayland) Nvidia card
I have 3 monitors - a 4k (125% scaling) monitor in the center and a 2k (100% scaling) monitor on each side
When opening on the center monitor it appears tiny, crushed, and finicky.
If I drag the window fully into one of my side monitors then it corrects itself, then after dragging it back to the center it stays corrected.
It used to ask me to restart for zoom changes to take effect, but after upgrading to dbeaver 26.1.3.202607191738 it no longer does.
